app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt

```diff
@@ -34,8 +34,8 @@ class SignUpFragment : Fragment() {
             signUp.firstName = firstNameText.text.toString()
             signUp.lastName = lastNameText.text.toString()
             confirmPassword = confirmPasswords.text.toString()
-
             signUpActivityViewModel.signUp(signUp, confirmPassword)
+            if(!signUpActivityViewModel.isNetworkConnected()) Utils.showNoInternetDialog(activity)
```

Comment:
This is not how you do it, subscribe to a livedata which pushes network not available errors. No ifs in activities